homeassistant/components/freebox/device_tracker.py

```diff
@@ -1,65 +1,150 @@
 """Support for Freebox devices (Freebox v6 and Freebox mini 4K)."""
-from collections import namedtuple
+from datetime import datetime
 import logging
+from typing import Dict
 
-from homeassistant.components.device_tracker import DeviceScanner
+from homeassistant.components.device_tracker import SOURCE_TYPE_ROUTER
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import HomeAssistantType
 
-from . import DATA_FREEBOX
+from .const import DEFAULT_DEVICE_NAME, DEVICE_ICONS, DOMAIN
+from .router import FreeboxRouter
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_get_scanner(hass, config):
-    """Validate the configuration and return a Freebox scanner."""
-    scanner = FreeboxDeviceScanner(hass.data[DATA_FREEBOX])
-    await scanner.async_connect()
-    return scanner if scanner.success_init else None
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Freebox component."""
+    router = hass.data[DOMAIN][entry.unique_id]
+    tracked = []
```

Comment:
A set is better when we expect the items of the container to be unique.
```suggestion
    tracked = set()
```